src/libs/OptionsListUtils.js

```diff
@@ -662,7 +663,94 @@ function hasEnabledOptions(options) {
 }
 
 /**
- * Build the options for the category tree hierarchy via indents
+ * Sorts categories using a simple object.
+ * It builds an hierarchy (based on an object), where each category has a name and other keys as subcategories.
+ * Via the hierarchy we avoid duplicating and sort categories one by one. Subcategories are being sorted alphabetically.
+ *
+ * @param {Object<String, {name: String, enabled: Boolean}>} categories
+ * @returns {Array<Object>}
+ */
+function sortCategories(categories) {
+    // Sorts categories alphabetically by name.
+    const sortedCategories = _.values(categories).sort((a, b) => {
+        if (a.name < b.name) {
+            return -1;
+        }
+
+        if (a.name > b.name) {
+            return 1;
+        }
+
+        return 0;
+    });
+
+    // An object that respects nesting of categories. Also, can contain only uniq categories.
+    const hierarchy = {};
+
+    /**
+     * Iterates over all categories to set each category in a proper place in hierarchy
+     * It gets a path based on a category name e.g. "Parent: Child: Subcategory" -> "Parent.Child.Subcategory".
+     * {
+     *   Parent: {
+     *     name: "Parent",
+     *     Child: {
+     *       name: "Child"
+     *       Subcategory: {
+     *         name: "Subcategory"
+     *       }
+     *     }
+     *   }
+     * }
+     */
+    _.each(sortedCategories, (category) => {
```

Comment:
You used `_.reduce` a few lines later, maybe this would be a bit cleaner if we used reduction instead of mutating the `hierarchy` object?

Reply:
Yes, I use the reduction to covert the `hierarchy` object into a sorted array. Before the reduction, we need do 2 actions:
1. Sort all categories alphabetically. It helps with top-level categories.
2. Build the `hierarchy` object. It helps sort subcategories, keep them uniquely and store initial names. 

So, I don't see a variant how to combine these 2 different functions.

Reply:
I just meant that the `hierarchy` itself could probably also be build in a mostly functionally-pure way

Reply:
Hmmm... Could you please share an example?
It is a simple loop right now, where the app sets values one by one. I don't see any concerns with functional programming patterns here.

Reply:
Okey, let's leave this. I agree that the code is acceptable as-is.